frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/ServerEmbedCodePane.tsx

```diff
@@ -0,0 +1,186 @@
+import _ from "underscore";
+import { t } from "ttag";
+import { getEmbedServerCodeExampleOptions } from "metabase/public/lib/code";
+import type {
+  EmbeddingDisplayOptions,
+  EmbeddingParameters,
+  EmbedResource,
+  EmbedResourceType,
+  ServerCodeSampleConfig,
+} from "metabase/public/lib/types";
+
+import "ace/mode-clojure";
+import "ace/mode-javascript";
+import "ace/mode-ruby";
+import "ace/mode-python";
+
+import { DEFAULT_DISPLAY_OPTIONS } from "./config";
+import { CodeSample } from "./CodeSample";
+
+type EmbedCodePaneProps = {
+  siteUrl: string;
+  secretKey: string;
+  variant: "overview" | "parameters" | "appearance";
+  resource: EmbedResource;
+  resourceType: EmbedResourceType;
+  params: EmbeddingParameters;
+  displayOptions: EmbeddingDisplayOptions;
+  initialPreviewParameters: EmbeddingParameters;
+
+  serverCodeOptions: ServerCodeSampleConfig[];
+  selectedServerCodeOptionName: string;
+  setSelectedServerCodeOptionName: (languageName: string) => void;
+
+  className?: string;
+};
+
+export const ServerEmbedCodePane = ({
+  siteUrl,
+  secretKey,
+  variant,
+  resource,
+  resourceType,
+  params,
+  displayOptions,
+  initialPreviewParameters,
+  serverCodeOptions,
+  selectedServerCodeOptionName,
+  setSelectedServerCodeOptionName,
+
+  className,
+}: EmbedCodePaneProps): JSX.Element | null => {
+  const selectedServerCodeOption = serverCodeOptions.find(
+    ({ name }) => name === selectedServerCodeOptionName,
+  );
+
+  if (!selectedServerCodeOption) {
+    return null;
+  }
+
+  const { hasParametersCodeDiff, hasAppearanceCodeDiff, highlightedText } =
+    getHighlightedCode({
+      initialPreviewParameters,
+      params,
+      selectedServerCodeOption,
+      selectedServerCodeOptionName,
+      siteUrl,
+      secretKey,
+      resourceType,
+      resource,
+      displayOptions,
+    });
+
+  if (variant === "overview") {
+    return (
+      <CodeSample
```

Comment:
Can we put the `CodeSample/title` values into a function and only call `CodeSample` once? Something like

```
const getTitleText = (variant) => {
	... put the three different titles here ...
}


// more things here

<CodeSample
        dataTestId="embed-backend"
        className={className}
        title={getTitleText()}
        selectedOptionName={selectedServerCodeOptionName}
        languageOptions={serverCodeOptions.map(({ name }) => name)}
        source={selectedServerCodeOption.source}
        textHighlightMode={selectedServerCodeOption.mode}
        highlightedText={highlightedText}
        onChangeOption={setSelectedServerCodeOptionName}
      />
``` 

since all of these calls seem the same